src/plugins/discover/public/context_awareness/profile_providers/register_profile_providers.ts

```diff
@@ -40,61 +39,62 @@ export const registerProfileProviders = ({
   const rootProfileProviders = createRootProfileProviders(providerServices);
   const dataSourceProfileProviders = createDataSourceProfileProviders(providerServices);
   const documentProfileProviders = createDocumentProfileProviders(providerServices);
-  const enabledProfileIds = uniq([
-    ...extractProfileIds(rootProfileProviders),
-    ...extractProfileIds(dataSourceProfileProviders),
-    ...extractProfileIds(documentProfileProviders),
-    ...experimentalProfileIds,
-  ]);
 
-  registerEnabledProfileProviders({
+  const enabledRootProfileProviders = rootProfileProviders.filter(
+    ({ isEnabled = true, profileId }) => isEnabled || experimentalProfileIds.includes(profileId)
+  );
+
+  const enabledDataSourceProfileProviders = dataSourceProfileProviders.filter(
+    ({ isEnabled = true, profileId }) => isEnabled || experimentalProfileIds.includes(profileId)
+  );
+
+  const enabledDocumentProfileProviders = documentProfileProviders.filter(
+    ({ isEnabled = true, profileId }) => isEnabled || experimentalProfileIds.includes(profileId)
+  );
```

Comment:
Maybe we could add this filtering logic directly to `registerProfileProvidersInternal` instead and pass down `experimentalProfileIds`.

Reply:
done here: 0739ef2f137